test/spec/modules/33acrossBidAdapter_spec.js

```diff
@@ -589,8 +601,110 @@ describe('33acrossBidAdapter:', function () {
         expect(builtServerRequests).to.deep.equal([serverRequest]);
       });
     });
+
+    context('when there is valid schain object in the bidRequest', function() {
+      it('builds request with schain info in source', function() {
+        const schain = {
+          'ver': "1.0",
+          'complete': 1,
+          'nodes': [
+            {
+              'asi': "bidderA.com",
+              'sid': "00001",
+              'hp': 1
+            }
+          ]
+        };
+
+        bidRequests[0].schain = schain;
+
+        const ttxRequest = new TtxRequestBuilder()
+          .withSchain(schain)
+          .build();
+        const serverRequest = new ServerRequestBuilder()
+          .withData(ttxRequest)
+          .build();
+
+        const builtServerRequests = spec.buildRequests(bidRequests, {});
+
+        expect(builtServerRequests).to.deep.equal([serverRequest]);
+
+      });
+    });
+
+    context('when there no schain object is passed', function() {
+      it('does not set source field', function() {
+        const ttxRequest = new TtxRequestBuilder()
+        .build();
+        
+        const serverRequest = new ServerRequestBuilder()
+        .withData(ttxRequest)
+        .build();
+
+        const builtServerRequests = spec.buildRequests(bidRequests, {});
+
+        expect(builtServerRequests).to.deep.equal([serverRequest]);
+      });
+    });
+
+    context('when invalid schain object is passed in the bidRequest', function() {
+      it('builds request with source set to null', function() {
+        const invalidSchainValues = [
```

Comment:
all these cases will failed on `complete: '1'`, it is a string not a number. 
also add a case for a number thats not 0|1

Reply:
Oops my bad about '1'.  Fixed it. 
1. The test with `complete:"1"` should address the logic that marks schain as invalid when it's not 0|1
2. I have a test case with `asi:""` which should address the logic that marks schain as invalid if it is falsey.